### ISSUE TITLE
407 Proxy Auth required

### ISSUE THREAD

**Issue by jonwitts** [unosquare/passcore#181]
**Describe the bug**
When submitting a password change we get the following error message pop-up:
"Response status code does not indicate success: 407 (Proxy Authentication Required)."

The server is configured to have direct out-going internet access bypassing the proxy for all sites, which has been confirmed with multiple browsers; however on submission of the password change page we see a 407 request to google.com attempting to use the proxy server.

If we disable Recaptcha in the appsettings.json we do not get the 407 error nor see any access to the proxy, however we get a pop-up saying ""Could not verify you are not a robot".

In both cases passwords are not changed.

**Comment by geoperez** [comment]
Hello,

For the issue related to the Proxy Authentication, I have no idea. You may need to check the logs.

Regarding the reCaptcha, you need to verify that `RecaptchaPrivateKey` is empty to bypass the check on backend.

**Comment by jonwitts** [comment]
Hi,

We have increased all three logs up to Debug level and cannot find anything in there about proxy auth errors. I can confirm that with the reCaptcha disabled and RecaptchaPrivateKey empty it appears to allow us to reset users passwords (waiting for the min password age to pass on our test user to confirm!).

We would quite like to enable the reCaptcha though... What method implements the reCaptcha object? Could there be something in there which is turning on auto-proxy detection?

Thanks,
Jon

**Comment by geoperez** [comment]
But your IIS server is accessing to the web by proxy or not?

**Comment by jonwitts** [comment]
No, we have set a bypass up so the server can get directly out to the internet. However if auto-detection were running in some of the scripts it is possible that it would detect our proxy config via the DNS WPAD entry...

**Comment by geoperez** [comment]
You may need to check the proxy configuration for web.config: https://docs.microsoft.com/en-us/dotnet/framework/network-programming/proxy-configuration

**Comment by geoperez** [comment]
Status?

**Comment by git0000** [comment]
I had the same problem with 407 Proxy Authentication
Did try put in the proxy configuration in the web.config which didn't help.

We have TMG in our DMZ which require form authentication, which I think is stopping Captcha going out/in somehow. However I can't change anything in the TMG.

While testing, in appsettings.json, I left
RecaptchaPrivateKey :"", //  <-- yes nothing in there
Recaptcha:{
 "IsEnabled": true,
 "SiteKey": "whateverisyours",
 "LanguageCode: "en"
},

To my amaze, it works! Captcha still working, and I could change password successfully.
Hope this help someone else.

Tested PassCore340 running on both Server2016 & Server 2008 successfully.
